server/text_generation_server/models/ct2_causal_lm.py

```diff
@@ -51,6 +60,434 @@
 tracer = trace.get_tracer(__name__)
 
 
+@dataclass
```

Comment:
nit, it looks like we tend to prefer to require keyword args for dataclasses (i.e. `kw_only=true`)